Merge pull request #1 from linkedin/master

shenodaguirguis opened the pull request:
sycn with upstream

wmoustafa reviewed (commented):
Is this a revert? Can we link to another commit/PR that is being reverted?

ljfgem commented:
According to the code change, I don't think it's the revert of these 2 commits.
@shenodaguirguis can you check?

shenodaguirguis commented:
@wmoustafa  @ljfgem  the diff seems to match that the "Merge branch 'master' into master" commit made these changes

`$ git diff 098f21d4f7086ec16d7120b5ddbc5381c7acecc3 7750fba29cec445cdc1f160d26f842624d4d2bf3 
diff --git a/coral-hive/src/main/java/com/linkedin/coral/hive/hive2rel/functions/StaticHiveFunctionRegistry.java b/coral-hive/src/main/java/com/linkedin/coral/hive/hive2rel/functions/StaticHiveFunctionRegistry.java
index 8a08787..d85deab 100644
--- a/coral-hive/src/main/java/com/linkedin/coral/hive/hive2rel/functions/StaticHiveFunctionRegistry.java
+++ b/coral-hive/src/main/java/com/linkedin/coral/hive/hive2rel/functions/StaticHiveFunctionRegistry.java
@@ -471,6 +471,10 @@ public class StaticHiveFunctionRegistry implements FunctionRegistry {
         FunctionReturnTypes.STRING, family(SqlTypeFamily.INTEGER));
     createAddUserDefinedFunction("com.linkedin.tscp.reporting.dali.udfs.SponsoredMessageNodeId", ReturnTypes.INTEGER,
         family(SqlTypeFamily.STRING));
+    createAddUserDefinedFunction("com.linkedin.orbit.emerger.coercerudfs.DynamicsLineOfBusinessCoercer",
+        FunctionReturnTypes.STRING, STRING);
+    createAddUserDefinedFunction("com.linkedin.etg.business.common.udfs.MapD365OptionSet", FunctionReturnTypes.STRING,
+        STRING_STRING_STRING);
 
     // The following UDFs are already defined using Transport UDF.
     // The class name is the corresponding Hive UDF.
@@ -534,7 +538,7 @@ public class StaticHiveFunctionRegistry implements FunctionRegistry {
         family(SqlTypeFamily.ANY, SqlTypeFamily.STRING));
     createAddUserDefinedFunction("com.linkedin.udfs.standard.hive.ObfuscateMapValEvolve", ARG0,
         family(SqlTypeFamily.ANY, SqlTypeFamily.STRING));
-
+    createAddUserDefinedFunction("com.linkedin.jobs.udf.hive.ConvertIndustryCode", FunctionReturnTypes.STRING, STRING);
     // This is a Hive Custom UDF which is a simplified version of 'date-converter' package.
     // This UDF is not converted to a transport UDF.
     createAddUserDefinedFunction("com.linkedin.dali.customudf.date.hive.DateFormatToEpoch", BIGINT_NULLABLE,
`

shenodaguirguis commented:
trino regression tests introduced one new unrelated failure https://ltx1-jenkins.grid.linkedin.com:8443/job/hadoop-dev/view/Dali/job/Dali-Pipeline-coral-tools-regression/353/execution/node/3/ws/coral-tools/coral-tools/regression/orbit_emerger_mp_versioned-ingest_d365_account_line_of_business_decorated_13_0_3.log

wmoustafa commented on coral-spark/src/main/java/com/linkedin/coral/spark/TransportableUDFMap.java:
```diff
@@ -137,10 +137,6 @@ private TransportableUDFMap() {
     add("com.linkedin.stdudfs.lookup.hive.BrowserLookup", "com.linkedin.stdudfs.lookup.spark.BrowserLookup",
         "ivy://com.linkedin.standard-udfs-parsing:parsing-stdudfs:3.0.3?classifier=spark_2.11",
         "ivy://com.linkedin.standard-udfs-parsing:parsing-stdudfs:3.0.3?classifier=spark_2.12");
-
-    add("com.linkedin.jobs.udf.hive.ConvertIndustryCode", "com.linkedin.jobs.udf.spark.ConvertIndustryCode",
-        "ivy://com.linkedin.jobs-udf:jobs-udfs:2.1.2?classifier=spark_2.11",
-        "ivy://com.linkedin.jobs-udf:jobs-udfs:2.1.2?classifier=spark_2.12");
```
Are we supposed to remove this code?

wmoustafa commented:
> @wmoustafa @ljfgem the diff seems to match that the "Merge branch 'master' into master" commit made these changes
> 
Did this change since I see the UDF changes are no longer part of the PR?

shenodaguirguis commented:
> > @wmoustafa @ljfgem the diff seems to match that the "Merge branch 'master' into master" commit made these changes
> 
> Did this change since I see the UDF changes are no longer part of the PR?

yes, I updated the PR and I reverted 2 commits only (skipped reverting the "merge Master into master" commit

wmoustafa commented:
Thanks! Will merge after integration tests.

shenodaguirguis commented:
iTests succeeded (Trino failed with unrelated error after successfully translating all views)
